ref/haskell/src/Codec/Binary/Bech32.hs

```diff
@@ -156,4 +177,14 @@ segwitDecode hrp addr = do
 segwitEncode :: HRP -> Word8 -> Data -> Maybe BS.ByteString
 segwitEncode hrp witver witprog = do
     guard $ segwitCheck witver witprog
-    bech32Encode hrp $ UnsafeWord5 witver : toBase32 witprog
+    rightToMaybe $ bech32Encode hrp $ UnsafeWord5 witver : toBase32 witprog
+
+rightToMaybe :: Either l r -> Maybe r
+rightToMaybe = either (const Nothing) Just
+
+maybeToRight :: l -> Maybe r -> Either l r
+maybeToRight l = maybe (Left l) Right
+
+verify :: Bool -> a -> Either a ()
```

Comment:
I feel that this would work better with the argument order reversed.  Then it would be similar to `maybeToRight`.